pkg/resource/replication_group/delta_util.go

```diff
@@ -91,6 +92,14 @@ func engineVersionsMatch(
 		return r.MatchString(latestEV)
 	}
 
+	// if the version is higher than 6, skip the upstream patch version when comparing.
+	// See https://github.com/aws-controllers-k8s/community/issues/1737
+	majorVersion, _ := strconv.Atoi(desiredEV[0:1])
+	if majorVersion >= 6 {
+		r, _ := regexp.Compile(desiredEV + ".*")
```

Comment:
regex compile is an expensive operation i wouldn't recommend calling everytime the reconciliation loop is called. How about declaring it outside?
Maybe even better, how about using a semver parsing library and just make sure major/minor versions match?

Reply:
I just saw that `regex.Compare` is above as well... ideally we should not.. (git blame 👀 )